docs/developing.md

```diff
@@ -57,3 +57,47 @@ $ cd <application project directory>
 $ yalc remove @hypothesis/frontend-shared
 $ yarn --check-files
 ```
+
+## Adding new components
+
+Before adding a component, determine what category of component it will be — simple, presentational or composite (details can be found in the Pattern Library) — and what group of components it belongs with (e.g. Layout, Data Display).
+
+Each component in this package should have:
+
+1. A component module in `src/components`
+2. A test module
+3. Pattern-library documentation in `src/pattern-library/components`
```

Comment:
As above, I think it would be useful to indicate the full directory structure here, which is currently `src/pattern-library/components/patterns/{group}/{ComponentName}Page.js`.